include/led.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_LED_H
+#define _ZEPHYR_LED_H
+
+/**
+ * @file
+ * @brief Public LED driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief Internal driver specific representation of an LED device
+ */
+
+struct led_data {
+	/* Minimum acceptable LED blinking time period (in ms) */
+	u32_t min_period;
+	/* Maximum acceptable LED blinking time period (in ms) */
+	u32_t max_period;
+	/* Minimum acceptable LED brightness value */
+	u16_t min_brightness;
+	/* Maximum acceptable LED brightness value */
+	u16_t max_brightness;
+};
+
+/**
+ * @typedef led_api_blink()
+ * @brief Callback API for blinking an LED
+ *
+ * @see led_blink() for argument descriptions.
+ */
+typedef int (*led_api_blink)(struct device *dev, u32_t led, u32_t period);
+
+/**
+ * @typedef led_api_set_brightness()
+ * @brief Callback API for setting brightness of an LED
+ *
+ * @see led_set_brightness() for argument descriptions.
+ */
+typedef int (*led_api_set_brightness)(struct device *dev, u32_t led,
+				      u16_t value);
+/**
+ * @typedef led_api_on()
+ * @brief Callback API for turning on an LED
+ *
+ * @see led_on() for argument descriptions.
+ */
+typedef int (*led_api_on)(struct device *dev, u32_t led);
+
+/**
+ * @typedef led_api_off()
+ * @brief Callback API for turning off an LED
+ *
+ * @see led_off() for argument descriptions.
+ */
+typedef int (*led_api_off)(struct device *dev, u32_t led);
+
+/**
+ * @brief LED driver API
+ *
+ * This is the mandatory API any LED driver needs to expose.
+ */
+struct led_driver_api {
+	led_api_blink blink;
+	led_api_set_brightness set_brightness;
+	led_api_on on;
+	led_api_off off;
+};
+
+/**
+ * @brief Blink an LED
+ *
+ * This routine starts blinking an LED forever with the given time period
+ *
+ * @param dev LED device
+ * @param led LED channel/pin
+ * @param period Time period (in milliseconds) an LED should blink
+ * @return 0 on success, negative on error
+ */
+static inline int led_blink(struct device *dev, u32_t led, u32_t period)
```

Comment:
The commit log says led_oneshot_blink() is also meant to be supported here, but I don't see it -- it's fine with me if you want to do that later, just making sure it wasn't forgotten by accident.

Reply:
Sorry, it is on my to-do list and not supported as of now. Will remove that from commit message.

Reply:
What would be the benefit of ON and OFF time than specifying the time period? Anyway we will be calculating the time period by calculating the sum of ON and OFF time in driver. IMO it will add unnecessary overhead in API.

Reply:
Did a bit of research and it looks like there are LED devices capable of setting the ON time and OFF time independently. Will remove `period` parameter and add `delay_on`,`delay_off`.

Thanks for pointing out :)